Algorithm.Python/ParameterizedAlgorithm.py

```diff
@@ -0,0 +1,75 @@
+﻿# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License"); 
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import clr
+clr.AddReference("System")
+clr.AddReference("QuantConnect.Algorithm")
+clr.AddReference("QuantConnect.Indicators")
+clr.AddReference("QuantConnect.Common")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Indicators import *
+
+
+class ParameterizedAlgorithm(QCAlgorithm):
+    def __init__(self):
+        # we place attributes on top of our fields or properties that should 
+        # receive their values from the job. The values 100 and 200 are just
+        # default values that or only used if the parameters do not exist
+        self.Fast = None
+        self.Slow = None
+        self.FastPeriod = 100
+        self.SlowPeriod = 200
+        self.GetParameters()
+
+
+    def GetParameters(self):
+        '''Get parameters from config.json'''
+        import json
```

Comment:
We found a [solution]( https://github.com/QuantConnect/Lean/blob/5b91a14685bb3ae092084a66049f848161de1d5f/Algorithm.Python/ParameterizedAlgorithm.py#L46) that calls `IAlgorithm.GetParameters` at `Initialize`.
We could put the .NET style attributes in Python in the Python TODO list together with the `QCPythonAlgorithm` base class.